drivers/pwm/pwm_nrfx.c

```diff
@@ -209,27 +217,27 @@ static int pwm_nrfx_set_cycles(const struct device *dev, uint32_t channel,
 		/* Don't wait here for the peripheral to actually stop. Instead,
 		 * ensure it is stopped before starting the next playback.
 		 */
-		nrfx_pwm_stop(&config->pwm, false);
-		data->stop_requested = true;
+		if (data->active && !data->stop_requested) {
+			data->stop_requested = true;
+			nrfx_pwm_stop(&config->pwm, false);
+		}
 	} else {
-		if (data->stop_requested) {
-			data->stop_requested = false;
-
-			/* After a stop is requested, the PWM peripheral stops
-			 * pulse generation at the end of the current period,
-			 * and till that moment, it ignores any start requests,
-			 * so ensure here that it is stopped.
-			 */
-			while (!nrfx_pwm_stopped_check(&config->pwm)) {
-			}
+		/* After a stop is requested, the PWM peripheral stops
+		 * pulse generation at the end of the current period,
+		 * and till that moment, it ignores any start requests,
+		 * so ensure here that it is stopped.
+		 */
+		while (data->stop_requested) {
```

Comment:
what about the case when we `nrfx_pwm_stop()` is called for the second time on already stopped pwm. There is an assert in nrfx_pwm.c. It can happen if 0/100 duty cycle is set second channel.

Reply:
a, ok. We can use `nrfx_pwm_stopped_check(&config->pwm)` there.

Reply:
> what about the case when we `nrfx_pwm_stop()` is called for the second time on already stopped pwm. There is an assert in nrfx_pwm.c. It can happen if 0/100 duty cycle is set second channel.

A second stop is not a problem. If you mean `NRFX_ASSERT(p_cb->state != NRFX_DRV_STATE_UNINITIALIZED);`, the condition is still fulfilled then. Actually, the existing code works this way - there's no protection against a second call to `nrfx_pwm_stop()`.

Reply:
in that case i simplified change and allow double nrfx_pwm_stop() call.